### PR TITLE
Split and simplify `fgMorphOneAsgBlockOp`

### PR DESCRIPTION
**Pull request by SingleAccretion**
The main goal of this change is to extract the parts of `fgMorphOneAsgBlockOp` that actually do useful work and delete the rest.

To that end:
1) Import `initobj <primitive>` as `stind <primitive zero>`.
2) Fix some regressions and gather some improvements by retyping certain FP stores to in-memory locals to integer ones. This is where the bulk of the diffs for this change come from. Regressions are due to alignment or the fact some code gets a bit larger (but avoids a load from the data section).
3) Delete code from `fgMorphOneAsgBlockOp`: extract the "init block" part to the general block morpher, leave the copy block part as-is (for now).

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib 

The CI is down, but this is otherwise ready for review.

**Comment by SingleAccretion** [comment]
There are some SIMD-related failures here getting rid of which is best done by waiting for #76745 to land.

**Comment by SingleAccretion** [comment]
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=68076&view=ms.vss-build-web.run-extensions-tab).

@dotnet/jit-contrib

File: src/coreclr/jit/lower.cpp [hunk 1]
```diff
@@ -7467,6 +7426,89 @@ bool Lowering::TryTransformStoreObjAsStoreInd(GenTreeBlk* blkNode)
...
+// TryRetypingFloatingPointStoreToIntegerStore: Retype an FP memory store.
+//
+// On some targets, integer stores are cheaper and/or smaller than their
+// floating-point counterparts, because, e. g., integer immediates can be
+// encoded inline while FP ones need to be loaded from the data section.
+//
+// Arguments:
+//    store - The store node
+//
+void Lowering::TryRetypingFloatingPointStoreToIntegerStore(GenTree* store)
```
Comment (SingleAccretion) [inline src/coreclr/jit/lower.cpp:7439]: The original retyping was done under MinOpts too, so this follows; there is of course an argument to be made it should not.

**Review by jakobbotsch — APPROVED** [review]

**Comment by SingleAccretion** [comment]
Looks like Windows ARM64 stress will time out.

Failures so far are known:
1) #78023 ([stress](https://dev.azure.com/dnceng-public/public/_build/results?buildId=76043&view=results))
2) #77241 ([libraries stress](https://dev.azure.com/dnceng-public/public/_build/results?buildId=76044&view=results))

**Pull request merged**